### ISSUE TITLE
Cant input asterisk

### ISSUE THREAD

**Issue by antosha417** [nvim-orgmode/orgmode#370]
### Describe the bug

When I'm trying to input an asterisk into `.org` file nightly neovim freezes. I tested it with neovim 7.2 and everything works fine there.

### Steps to reproduce

1. `nvim -u minimal_init.lua test.org`
2. go to insert mode
3. press `*`
4. neovim freezes

### Expected behavior

I'm able to input a star

### Minimal init.lua

https://github.com/nvim-orgmode/orgmode/blob/master/scripts/minimal_init.lua will do

### OS / Distro

macOS 12.3

### Neovim version/commit

v0.8.0-dev-nightly

**Comment by Maltimore** [comment]
I have the same thing. Not only that, but I observed that neovim starts eating up memory - about 0.5 GB of RAM per second in my case. If I wait long enough before killing it (only `kill -9` works, normal `kill` isn't enough), my entire operating system becomes unusable.

More information I can add to this issue is that I'm on Ubuntu (not MacOS like the original poster) and I tried alacritty and konsole and with and without tmux, happens every time. If I have it running within tmux, and I kill the entire tmux pane, it doesn't help, nvim doesn't get killed.
